### ISSUE TITLE
simulators not building libboost-locale

### ISSUE THREAD

**Issue by tumnus** [apotocki/boost-iosx#16]
Hi - love this project. For some reason, when I run the build script, I get all the libraries except "locale":

`fatal error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: can't open input file: stage/iossim-arm64/lib/libboost_locale.a (No such file or directory)`

To speed up debugging...

1. I set `LIBS_TO_BUILD="--with-locale"` to only build the problem lib.
2. I commented out all the `build_xcframework boost_<name>` lines at the end of the script *except* for the `build_xcframework boost_locale` because if I did not, they would all fail when they encountered missing libraries not built.

The build runs through until I get the same result as above. Next, I listed the staging directories to see what had been built:

```
1.85.0 $ ls boost/stage/*/lib
boost/stage/catalyst-arm64/lib:
cmake			libboost_chrono.a	libboost_cobalt.a	libboost_locale.a	libboost_thread.a

boost/stage/catalyst-x86_64/lib:
cmake			libboost_chrono.a	libboost_cobalt.a	libboost_locale.a	libboost_thread.a

boost/stage/catalyst/lib:
libboost_locale.a

boost/stage/ios/lib:
cmake			libboost_chrono.a	libboost_cobalt.a	libboost_locale.a	libboost_thread.a

boost/stage/iossim-arm64/lib:
cmake			libboost_cobalt.a

boost/stage/iossim-x86_64/lib:
cmake			libboost_cobalt.a

boost/stage/iossim/lib:

boost/stage/macosx-arm64/lib:
cmake			libboost_chrono.a	libboost_cobalt.a	libboost_locale.a	libboost_thread.a

boost/stage/macosx-x86_64/lib:
cmake			libboost_chrono.a	libboost_cobalt.a	libboost_locale.a	libboost_thread.a

boost/stage/macosx/lib:
libboost_locale.a

boost/stage/xros/lib:
cmake			libboost_chrono.a	libboost_cobalt.a	libboost_locale.a	libboost_thread.a

boost/stage/xrossim-arm64/lib:
cmake			libboost_cobalt.a

boost/stage/xrossim-x86_64/lib:
cmake			libboost_cobalt.a

boost/stage/xrossim/lib:
```

Obviously there is some logic to turn on some extra libs (no problem) but you can see the simulator directories are missing "locale".

Everything builds perfectly if I start with a fresh pull and remove "--with-locale" (and the corresponding `build_xcframework boost_locale` call).

Can I help troubleshoot this? Do you have any suggestions I can try? Is there some more info I can give you to get to the bottom of it? Would love to contribute to this brilliant project!

---- edit -----

I;ve just spotted this:

```
    - icu                      : no [2]
    - iconv (libc)             : no [2]
    - iconv (separate)         : no [2]
- Boost.Locale needs either iconv or ICU library to be built.
    - icu                      : no [3]
    - iconv (libc)             : no [3]
    - iconv (separate)         : no [3]
- Boost.Locale needs either iconv or ICU library to be built.
    - cxx20_hdr_concepts       : yes [2]
    - cpp_lib_memory_resource  : no [2]
```

... so I'm wondering if it is not picking up the Pod

**Comment by apotocki** [comment]
It was a problem related to the latest XCode, but I've fixed it for 1.85.0 and 1.86.0 branches. Have you pulled the fix?

**Comment by tumnus** [comment]
Duh - yep, that worked. Thank you!
